File: 品味Java的21種設計模式/dp_cpp/iterator/example3/PayModel.cpp
#include "PayModel.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace iterator
			{
				namespace example3
				{

					std::string PayModel::getUserName()
					{
						return userName;
					}

					void PayModel::setUserName(std::string userName)
					{
						this->userName = userName;
					}

					double PayModel::getPay()
					{
						return pay;
					}

					void PayModel::setPay(double pay)
					{
						this->pay = pay;
					}

					std::string PayModel::ToString()
					{
						return "userName="+userName+",pay="+pay;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/iterator/example4/PayModel.java
package cn.javass.dp.iterator.example4;

/**
 * 工资描述模型对象
 */
public class PayModel {
    /**
     * 支付工资的人员
     */
    private String userName;
    /**
     * 支付的工资数额
     */
    private double pay;
    public String getUserName() {
        return userName;
    }
    public void setUserName(String userName) {
        this.userName = userName;
    }
    public double getPay() {
        return pay;
    }
    public void setPay(double pay) {
        this.pay = pay;
    }
    public String toString() {
        return "userName=" + userName + ",pay=" + pay;
    }
}
